src/hooks/usePipelineRunsForApplication.ts

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { useNamespace } from '../components/NamespacedPage/NamespacedPage';
+import { ComponentGroupVersionKind } from '../models';
+import { PipelineRunGroupVersionKind } from '../shared';
+import { PipelineRunKind } from '../shared/components/pipeline-run-logs/types';
+import { ComponentKind } from '../types';
+import { BUILD_APPLICATION_LABEL, BUILD_COMPONENT_LABEL } from '../utils/const';
+
+export const useLatestPipelineRunForComponent = (
+  component: ComponentKind,
+): PipelineRunKind | undefined => {
+  const { namespace } = useNamespace();
+  const [pipelineRuns, prLoaded, prError] = useK8sWatchResource<PipelineRunKind[]>({
+    groupVersionKind: PipelineRunGroupVersionKind,
+    namespace,
+    isList: true,
+  });
+  return React.useMemo(() => {
+    if (prLoaded && !prError) {
+      return pipelineRuns
+        ?.filter?.(
+          (plr) =>
+            plr.metadata.labels[BUILD_COMPONENT_LABEL] === component.metadata.name &&
+            plr.metadata.labels[BUILD_APPLICATION_LABEL] === component.spec.application,
```

Comment:
Is it possible to use matchLabel to filter in the watch hook itself?

Reply:
Yes, it should be possible to use label selectors now. Core team fixed that bug.

Reply:
It doesn't seem to be working If I add matchLabels it returns empty array